File: UI/pc-part-picker/src/model/Configuration.ts
import { Cpucooler } from "./Cpucooler";
import { Harddrive } from "./Harddrive";
import { Memory } from "./Memory";
import { Motherboard } from "./Motherboard";
import { Pccase } from "./Pccase";
import { Powersupply } from "./Powersupply";
import { Processor } from "./Processor";
import { Videocard } from "./Videocard";

export interface Configuration {
    processor?: Processor ;
    motherboard?: Motherboard;
    videocard?: Videocard ;
    memory?: Memory;
    harddrive?: Harddrive;
    pccase?: Pccase ;
    powersupply?: Powersupply;
    cpucooler?: Cpucooler;

    [key: string]: Processor
                  | Motherboard
                  | Videocard
                  | Memory
                  | Powersupply
                  | Pccase
                  | Harddrive
                  | Cpucooler
                  | undefined;
}

File: API/PCPartPicker/Models/Configuration.cs
using System;
using System.Collections.Generic;

namespace PCPartPicker.Models;

public partial class Configuration
{
    public int Id { get; set; }

    public string UserId { get; set; } = default!;
    public AppUser User { get; set; } = default!;
    public string Name { get; set; } = string.Empty;

    public int? MotherboardId { get; set; }

    public int? ProcessorId { get; set; }

    public int? VideocardId { get; set; }

    public int? MemoryId { get; set; }

    public int? PowersupplyId { get; set; }

    public int? CaseId { get; set; }

    public int? StorageId { get; set; }

    public int? CoolerId { get; set; }

    public virtual Pccase? Case { get; set; }

    public virtual Cpucooler? Cooler { get; set; }

    public virtual Memory? Memory { get; set; }

    public virtual Motherboard? Motherboard { get; set; }

    public virtual Powersupply? Powersupply { get; set; }

    public virtual Processor? Processor { get; set; }

    public virtual Harddrive? Storage { get; set; }


    public virtual Videocard? Videocard { get; set; }
}
